File: tunnel-common/src/test/java/com/tuuzed/tunnel/common/util/PortUtilsTest.java
package com.tuuzed.tunnel.common.util;

import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class PortUtilsTest {

    @Test
    public void inPortRule() {
        assertFalse(PortUtils.inPortRule("10000-21000", 40000));
        assertFalse(PortUtils.inPortRule("10000-21000,30000", 40000));
        assertFalse(PortUtils.inPortRule("10000-21000,30000", 1024));
        assertTrue(PortUtils.inPortRule("10000-21000,30000", 10002));
        assertTrue(PortUtils.inPortRule("10000-21000,30000", 10000));
        assertTrue(PortUtils.inPortRule("10000-21000,30000", 21000));
        assertTrue(PortUtils.inPortRule("10000-21000,30000", 30000));
    }
}

File: core/base/src/test/kotlin/lighttunnel/base/utils/PortUtilsTest.kt
package lighttunnel.base.utils

import org.junit.Test

class PortUtilsTest {

    @Test
    fun getRandomPort() {
        for (i in 1..10000) {
            val port = PortUtils.getAvailableTcpPort("4000-21000,30000,30001,30003")
            print("$port,")
            if (i != 0 && i % 10 == 0) {
                println()
            }
        }
    }

}
